pkg/golang/binary/parse.go

```diff
@@ -57,11 +63,38 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 			mod = dep.Replace
 		}
 
+		if !checkBuildID {
+			// get build id
+			buildID, err = getBuildID(r)
```

Comment:
Why not calculate this outside the loop ?

Reply:
Trivy skips binaries in the sbom which don't have any dep, so intend to skip computing build id in those cases

Reply:
Update